srv/salt/_modules/rgw.py

```diff
@@ -13,8 +19,88 @@ def configurations():
     """
     if 'roles' in __pillar__:
         if 'rgw_configurations' in __pillar__:
+            log.info("rgw_c: {}".format(__pillar__['rgw_configurations']))
             return list(Set(__pillar__['rgw_configurations']) &
                         Set(__pillar__['roles']))
         if 'rgw' in __pillar__['roles']:
             return [ 'rgw' ]
     return []
+
+
+def configuration(role):
+    """
+    Return the equivalent rgw role for the ganesha role. For instance,
+    the ganesha roles silver and silver-common will both return silver.
+    """
+    if role == 'ganesha':
+        role = 'rgw'
+    if 'roles' in __pillar__:
+        if 'rgw_configurations' in __pillar__:
+            for rgw_config in  __pillar__['rgw_configurations'].keys():
+                if rgw_config in role:
+                    return rgw_config
+    return 
+
+
+
+def users(role):
+    """
+    In progress...
+    """
+    if 'roles' in __pillar__:
+        if 'rgw_configurations' in __pillar__:
+            if role == 'ganesha' or role == 'rgw':
+                # Special case for default names
+                users = [ u['name'] for u in __pillar__['rgw_configurations']['rgw']['users'] ]
+                log.info("users: {}".format(users))
+                return users
+            if role in __pillar__['rgw_configurations']:
+                
+                return list(Set(__pillar__['rgw_configurations']) &
+                            Set(__pillar__['roles']))
+        if 'rgw' in __pillar__['roles']:
+            return []
+    return []
+
+def add_users(pathname="/srv/salt/ceph/rgw/cache"):
+    """
+    Write each user to its own file
+    """
+    for role in __pillar__['rgw_configurations']:
+        for user in __pillar__['rgw_configurations'][role]['users']:
+            command = "radosgw-admin user create --uid={} --display-name={} --email={}".format(user['name'], user['display'], user['email'])
```

Comment:
Need to modify this to take up additional and optional parameters like `email` `system` and `access` and `secret` flag, I'll rework this slightly